release notes/v0.35.0.md

```diff
@@ -0,0 +1,94 @@
+k6 v0.35.0 is here! :tada: It introduces several new features that nicely enhance usage of k6 as well as contains a whole number of fixes and ongoing efforts with refactoring.
+
+In total, we have closed 14 issues and merged 43 PRs. We have also branched out three new xk6 extensions during this release :star:
+
+## New features
+
+### VU tags ([#2172](https://github.com/grafana/k6/pull/2172))
+
+k6 now supports setting tags for VUs as part of the Execution API with an easy key-value interface. Example usage:
+
+```js
+import exec from 'k6/execution';
+import { sleep } from 'k6';
+
+export const options = {
+    duration: '10s',
+    vus: 3,
+};
+
+export default function () {
+    exec.vu.tags['mytag'] = 'value';
+    exec.vu.tags['vuId'] = exec.vu.idInTest;
+    console.log(`mytag is ${exec.vu.tags['mytag']} and my VU's ID in tags ${exec.vu.tags['vuId']}`);
+    sleep(5);
+}
+```
+
+There are some caveats, however; using `Object` or `Array` as a value will result in a warning or an exception if `--throw` option is used. Additionally, given that k6 has a whole bunch of system tags, one should be careful with using them as keys. You can read complete information about VU tags in [`k6/execution docs`](https://k6.io/docs/javascript-api/k6-execution/#vu).
```

Comment:
Hm, I wanted to avoid repeating the docs too much here but still make a warning... But listing only supported types is probably a good middle-ground. So it'll be smth like this:
```
There are some caveats, however: values can be only of `String`, `Number` or `Boolean` type, while values of other types will result either in a warning or an exception. Additionally, given that k6 has a whole bunch of system tags, one should be careful with using them as keys. You can read complete information about VU tags in [`k6/execution docs`](https://k6.io/docs/javascript-api/k6-execution/#vu).
```